harmony/harmony.py

```diff
@@ -364,8 +376,150 @@ def progress(self, job_id: str) -> int:
         can't be reached.
         """
         session = self._session()
-        response = session.get(self._status_url(job_id)).result()
+        response = session.get(self._status_url(job_id))
         if response.ok:
             return int((response.json())['progress'])
         else:
             response.raise_for_status()
+
+    def wait_for_processing(self, job_id: str, show_progress: bool = False) -> None:
+        """Retrieve a submitted job's completion status in percent.
+
+        Args:
+            job_id: UUID string for the job you wish to interrogate.
+
+        Returns:
+            The job's processing progress as a percentage.
+
+        :raises Exception: This can happen if an invalid job_id is provided or Harmony services
+        can't be reached.
+        """
+        check_interval = 3.0  # in seconds
+        ui_update_interval = 0.33  # in seconds
+        intervals = int(check_interval / ui_update_interval)
+        if show_progress:
+            with progressbar.ProgressBar(max_value=100, widgets=progressbar_widgets) as bar:
+                progress = None
+                check_cycle = cycle([True] + list(repeat(False, intervals - 1)))
```

Comment:
I could use some explanation on this one (maybe a comment).  Why does it only check every nth time around the loop?

Reply:
Rereading, I'm pretty sure I get it, but I think it could be more clear.  Maybe something like:

```python
progress = self.progress(job_id)
# Update the UI `intervals` times for each status fetch
for i in range(intervals):
   bar.update(progress)
   sys.stdout.flush()
   if progress >= 100: # or an error. see other comments
       break
   else:
       time.sleep(ui_update_interval)
```

Reply:
ProgressBar has a "spinner" UI element that changes on each update. There are two update intervals we want to work with: how often to fetch results from the server and how often to update the UI. We could make them the same interval but the spinner wouldn't move very much. It has a better aesthetic when it looks like it's more active or 'working more'. So I went with the smaller interval of the two and paired that with a sequence (cycle) of booleans. On True, do both UI updating and server checking. On False, only do UI updating. There's certainly other ways to do this sort of thing like keep track of timestamps and do server updates after a threshold period of time; reset threshold start time and do again, etc.

Reply:
I think we wrote the last two comments at the same time  : ) Sure - I'll get that updated.

Reply:
Updated code to be cleaner / simpler.